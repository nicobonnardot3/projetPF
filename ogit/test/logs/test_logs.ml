open Ogitlib
open Logs

(* on se place dans le r√©pertoire repo/ *)
let repo_root = "../../../../../repo"
let () = 
  Sys.chdir repo_root

(* on prepare le repertoire .ogit *)
let _ = Sys.command "rm -rf .ogit"
let _ = Sys.command "cp -r ../repo.origin/.ogit ."

(* test de date_fm *)
let _ = Unix.time ()
let date_bidon = fst (Unix.mktime { 
  Unix.tm_hour = 16; 
  Unix.tm_min = 8; 
  Unix.tm_sec = 40;
  Unix.tm_mday = 7; 
  Unix.tm_mon = 10 (*11 - 1*); 
  Unix.tm_year = 122 (* 2022 - 1900 *);
  Unix.tm_wday = 0; Unix.tm_yday = 0; Unix.tm_isdst = false
})

let () = Format.printf "date_fm date_bidon = %s@."  (date_fm date_bidon)


(* test de get_head et set_head *)

let head_bidon = [
  Digest.string "hello";
  Digest.string "world"
]

let head_bidon2 = 
  set_head head_bidon;
  get_head ()

let () = Format.printf "head_bidon = head_bidon2 ? %b@." (head_bidon = head_bidon2)


(* test de read_commit et store_commit *)

let hash_bidon = Digest.string "bidon"
let commit_bidon = {
  parents = head_bidon;
  date = date_bidon;
  message = "ceci est un commit bidon";
  content = hash_bidon
}

let hash_du_fichier_log = store_commit commit_bidon
let commit_bidon2 = read_commit hash_du_fichier_log

let () = Format.printf "commit_bidon = commit_bidon2 ? %b@." (commit_bidon = commit_bidon2)