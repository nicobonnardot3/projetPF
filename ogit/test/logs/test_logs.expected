date_fm date_bidon = 16:08:40-07/11/2022
head_bidon = head_bidon2 ? true
commit_bidon = commit_bidon2 ? true
